qiskit/providers/basicaer/qasm_simulator.py

```diff
@@ -410,11 +418,6 @@ def run(self, qobj, **backend_options):
             qobj = assemble(qobj, self, **out_options)
             qobj_options = qobj.config
         else:
-            warnings.warn(
-                "Using a qobj for run() is deprecated and will be removed in a future release.",
-                PendingDeprecationWarning,
-                stacklevel=2,
-            )
             qobj_options = qobj.config
```

Comment:
The PR doesn't need changing about this, I'm just musing: a potential downside of moving the deprecations to a decorator is that it becomes harder to spot the code changes necessary when the deprecation expires and we effect the removal.  Maybe we ought to think about asking people to put in a comment when adding a decorator if only part of the function would need removing?